src/components/dashboard/FeedList.js

```diff
@@ -101,11 +101,22 @@ const FeedList = ({
         try {
           userStorage.cancelOTPLEvent(id)
           goodWallet.cancelOTLByTransactionHash(id).catch(e => {
-            showErrorDialog('Canceling the payment link has failed', e)
+            log.error('cancel payment failed - quick actions', {
+              itemId: id,
+              message: e.message,
+              error: e,
+            })
             userStorage.updateOTPLEventStatus(id, 'pending')
+            showErrorDialog('Canceling the payment link has failed', e.message)
           })
         } catch (e) {
-          showErrorDialog('Canceling the payment link has failed', e)
+          log.error('cancel payment failed - quick actions', {
+            itemId: id,
+            message: e.message,
+            error: e,
+          })
+          userStorage.updateOTPLEventStatus(id, 'pending')
```

Comment:
why here we use updateOTPL and there recoverEvent?

Reply:
@sirpy I don't know why. It was written long time ago by other developers. I just used it for the catch clause.

Reply:
both places for canceling otpl should use same code

Reply:
Done